### PR TITLE
Automatic PyPI upload on tag using GitHub Actions

### PR DESCRIPTION
**Pull request by nsoranzo**
@martenson has already added a PyPI API token to this repository encrypted secrets, as explained in https://help.github.com/en/actions/automating-your-workflow-with-github-actions/creating-and-using-encrypted-secrets

### CODE REVIEW

**Comment by martenson** [comment]
The secret has been added to this repository.

**Comment by jmchilton** [comment]
Does https://github.com/galaxyproject/planemo/blob/master/docs/developing.rst need to be updated for these changes?

**Comment by nsoranzo** [comment]
> Does https://github.com/galaxyproject/planemo/blob/master/docs/developing.rst [](/galaxyproject/planemo/blob/HEAD@{2020-02-03T14:55:33Z}/docs/developing.rst) need to be updated for these changes?

I have updated the checklist now.

File: docs/developing.rst [hunk 1]
```diff
@@ -4,43 +4,18 @@ Release Checklist
...
-  * ``make push-release``
-  * ``make release-brew``
+* Commit outstanding changes.
+* Update version and history, commit, add tag, mint a new version and push
+  everything upstream with ``make release``
+* The new tag should automatically push the new release to PyPI via the
+  ``deploy`` job of the GitHub Actions workflow defined in
+  ``.github/workflows/ci.yaml`` .
+  If this didn't work, you can ``git checkout`` the tag and push to PyPI by
+  executing ``make release-artifacts``
```
Comment (jmchilton) [inline docs/developing.rst]: Awesome sauce, thanks for clarifying the process and how handle failures even!

Comment (nsoranzo) [reply docs/developing.rst]: Well, I'm not sure the upload will work the first time, so better plan ahead :laughing:

**Pull request merged**